### ISSUE TITLE
Caddy service fails to start

### ISSUE THREAD

**Issue by necrogami** [Homebrew/brew#10438]
# Bug report

**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew config` and `brew doctor` and included their output with your issue?

## What you were trying to do (and why)

run caddy server on brew services on osx Big Sur

## What happened (include command output)

<details>
  <summary>sudo brew services start caddy</summary>
  <pre>
Warning: Taking root:admin ownership of some caddy paths:
  /usr/local/Cellar/caddy/2.3.0/bin
  /usr/local/Cellar/caddy/2.3.0/bin/caddy
  /usr/local/opt/caddy
  /usr/local/opt/caddy/bin
  /usr/local/var/homebrew/linked/caddy
This will require manual removal of these paths using `sudo rm` on
brew upgrade/reinstall/uninstall.
Warning: caddy must be run as non-root to start at user login!
==> Successfully started `caddy` (label: homebrew.mxcl.caddy)
  </pre>
</details>
<details>
<summary>cat /usr/local/var/log/caddy.log</summary>
<pre>
{"level":"warn","ts":1611740307.688664,"msg":"unable to determine directory for user configuration; falling back to current directory","error":"$HOME is not defined"}
{"level":"info","ts":1611740307.709192,"msg":"using provided configuration","config_file":"/usr/local/etc/Caddyfile","config_adapter":""}
{"level":"info","ts":1611740307.710888,"logger":"admin","msg":"admin endpoint started","address":"tcp/localhost:2019","enforce_origin":false,"origins":["localhost:2019","[::1]:2019","127.0.0.1:2019"]}
{"level":"info","ts":1611740307.711073,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc0003771f0"}
{"level":"info","ts":1611740307.719483,"logger":"tls.cache.maintenance","msg":"stopped background certificate maintenance","cache":"0xc0003771f0"}
run: loading initial config: loading new config: loading http app module: provision http: getting tls app: loading tls app module: provision tls: provisioning automation policy 0: loading TLS automation management module: position 0: loading module 'internal': provision tls.issuance.internal: loading pki app module: provision pki: provisioning CA 'local': generating root: saving root certificate: mkdir caddy: read-only file system
</pre>
</details>

## What you expected to happen

I expect caddy to start as a service.

## Step-by-step reproduction instructions (by running `brew` commands)

`brew install caddy`
`sudo brew service start caddy`

## Output of `brew config` and `brew doctor` commands

<pre>
toast % brew config
HOMEBREW_VERSION: 2.7.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: bca4804a9e48de5319383d3eddadaa7f054c77da
Last commit: 12 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 2b73acc4463be52b42481656f69f9a89366daee1
Core tap last commit: 83 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: code
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_ANALYTICS: set
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.1-x86_64
CLT: 12.3.0.0.1.1607026830
Xcode: N/A
</pre>
<pre>
toast % brew doctor
Your system is ready to brew.
</pre>

**Comment by necrogami** [comment]
So it seems that when caddy fails to find working directory, or home it reverts to `/System/Volumes/Data` which is a read only directory and cannot write to it.

Do note I'm not sure if this matters or not however this was done with this caddy file:

<details>
<summary>cat /usr/local/etc/Caddyfile</summary>
<pre>
https://*.local.dev {
    root * /Users/necrogami/code/{labels.2}
    tls internal
    php_fastcgi 127.0.0.1:9000
}
</pre>
</details>

Basically it's setup to use caddy's internal self generated certificate authority.
